### ISSUE TITLE
Use material design standard grayscale to optimize UI text visibility

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#9125]
In comparison to several other outdoor apps we still produce sub-par contrast and sub-optimum visibility in harsh light conditions, in particular in our dark theme.

After some research it appears that the Android material design colors (https://stackoverflow.com/questions/36915508/what-are-the-material-design-dark-theme-colors) would produce a significant enhancement. In particular, we should apply the correction to
```
text_color_primary_dark
text_color_secondary_dark
text_color_tertiary_dark
divider_color_dark
activity_background_color_dark
activity_background_transparent_color_dark
list_background_color_dark
app_bar_color_dark
status_bar_color_dark
```
and all related icon colors.

I can provide PR if desired.

**Comment by dmpr0** [comment]
Hello! 
We use our own theme colors. Our night theme is first of all for night use, so light colors not bright.

**Comment by sonora** [comment]
Hi Dmitriy, yes, I know, but I am suggesting we reconsider this. Let me please clarify the following:
- I am only talking about the grayscale, i.e. our **text** screens (menus / dialogs / action screens)
- I am not talking about our general color palette (blue/orange) at all, and of course also not about the map
- The dark theme is popular because it provides better text readability in very sunny settings, (and on AMOLED screens it also saves power, often a concern in the backcountry). Thus this issue is not about night mode, really: Text readability is key here..
- But even at night: When I look at my Android 10 (set to dark theme) system setting screens, I have to say there is zero risk of being blinded at night., I think that is simply not a concern. And the readability of secondary text and explanations is superior to OsmAnd.

I invite us to test, or simply make a mental comparison with how google handles it. I am afraid they have the edge here ... ;)

**Comment by sonora** [comment]
Here's what I mean: It may not look like much, but please view these screens on your device in sunlight, and evaluate the readability of both headlines and explanatory text (screenshots are reduced in size so they fit side by side, hence loss in sharpness on both):

![Screenshot_20200603-214023_OsmAnd~_](https://user-images.githubusercontent.com/845735/83682099-bae5c100-a5e3-11ea-89a1-be2b812199e2.jpg) ![Screenshot_20200603-213925_Settings_](https://user-images.githubusercontent.com/845735/83682103-bc16ee00-a5e3-11ea-8c1c-ab187e59c307.jpg)

**Comment by dmpr0** [comment]
On this screen, uses wrong background color.
It use list_background_color_dark – https://github.com/osmandapp/Osmand/blob/master/OsmAnd/res/values/colors.xml#L19:L19, but must use activity_background_color_dark – https://github.com/osmandapp/Osmand/blob/master/OsmAnd/res/values/colors.xml#L15:L15

**Comment by sonora** [comment]
OK, but it clearly goes beyond just the background: The text readability in the material screenshot is also largely a result oft the text (foreground) colors. I suggest we

- simplify our setup by reducing the color constants we use to the pattern defined in the material scheme. It will be easier to consistently use for all our coders!
- And then why not also simply set them to exactly their grayscale values?

**Comment by sonora** [comment]
@dmpr0 , @vshcherb We have recently picked up work on colors (https://github.com/osmandapp/OsmAnd/commit/e77845314816bb3c20cf1ccc76589f8f63d62593). Should we maybe now use the occasion to remove some of the poor contrast situations we have in our UI palette?  I propose that for an outdoor app tackling this some time soon seems cardinal?

To assist with this,I have found this **Color Contrast Checker** https://imagecolorpicker.com/color-code, and here are some of its rating for our current color scheme. I propose anything which containing the word "poor" is a candidate for tuning a bit?

Background | Text | Contrast Value | Rating
-- | -- | -- | --
activity_background_color_light | text_color_primary_light | 14.13 | Super
  | text_color_secondary_light | 4.22 | Poor
  | text_color_tertiary_light | 1.61 | Very poor
activity_background_color_dark | text_color_primary_dark | 11.67 | Very Good
  | text_color_secondary_dark | 3.89 | Poor
  | text_color_tertiary_dark | 1.81 | Very Poor
</body>
</html>

**Comment by vshcherb** [comment]
Yes, that's I think the reason to have it sorted by contrast, what do you appeal than? If we talk about accessibility it's one thing, if we talk about normal people I think 4 Contrast very visible though it's marked as Poor

**Comment by sonora** [comment]
Yes, 4 is generally considered quite acceptable. But we could perhaps do better without losing anything - after all OsmAnd has major use cases beyond your normal "indoor / office use" apps. :wink:

If we raised our contrast some for all text/background cases we have, it could be a win for the accessibility as well as the bright outdoor light use cases.

I briefly checked Samsung's out of the box stock (dark) scheme, and find they use 17.18 as their top value, and find no worse than 2.73 (our tertiary would be 1.81 ?).

**Comment by vshcherb** [comment]
I will let @dmpr0 answer though in my experience even color blind people are pretty with contrast may be even better

**Comment by vshcherb** [comment]
@sonora if you convert colors to LAB color space and you have >= 5 by L value then it's usually more than enough.

**Comment by tomFlowee** [comment]
relevant issue: https://github.com/osmandapp/OsmAnd/issues/20738

Fun fact,

I posted the first screenshot on telegram and people didn't even find the text in the screenshot since the contrast is so poor.

i'm not sure if the less-than-4 is still accurate for todays OSMAnd, but the fact is that it is definitely too low contrast in the real world.

**Comment by XandrexOSM** [comment]
hello,

regarding "Material Design" by Google, I noticed that in october 2021, version 3 was released. It seems it is not the default version even today. Maybe it has settings that could allow OsmAnd to use it and maintain high visiibility for maps at night ?

on a side note, I have a question: does anyone knows if Google Maps and Waze use Material Design 3 ? maybe @sonora ?
I guess as map apps, they also want to achieve the same readability objective at night.

**Comment by sonora** [comment]
@XandrexOSM From what I see, and without detailed checking, the Android Google Maps app seems to pretty consistently always use the latest available version of OS design principles and colors. But please note that this is for the _user interface of the app_, I think map colors is an entirely different animal out of scope for this issue here. :wink:

**Comment by sonora** [comment]
PS: See https://github.com/osmandapp/OsmAnd/issues/20738#issuecomment-2332192896. Noticeably better by my testing! Wish we could expand this a bit like I describe.

**Comment by git2fa** [comment]
Hello developers, this bug is from Jun 3, 2020 without any solution!

OsmAnd is the **best APP for Outdoor activity** and there is only this little problem with the colors of the user interface.
**There must be a high contrast when it is used in the sun outside.**

Please solve this little problem, because it is not possible to work with such buttons in the sun.

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/a1231766-fb81-4b97-b3f5-34300edffcec"  width="300" height="auto"/> </td>
  </tr>
</table>

The design looks nice when there is less surrounding light, but design is not so important as functionality.

Using currently the recent Version 4.9.10 (491003) from https://f-droid.org/de/packages/net.osmand.plus/

**Comment by cyclingcat** [comment]
As someone who is heavily affected by this issue (= bad eyesight and getting worse with increasing age, including decreasing ability to read low-contrast texts):

Is there at least a temporary workaround/hack until this issue is resolved? Maybe some resource file which defines the colors and could be patched by the user, either directly on the device or by downloading to a desktop PC and uploading an edited version?

The cycling cat

**Comment by dmpr0** [comment]
@cyclingcat Hi! Those screenshots are outdated; the contrast has been increased. Android has an option to change the text size or increase its contrast.

**Comment by DmitryAlexei** [comment]
Please update your app to the latest version and adjust the background opacity of Zoom +- buttons in 
- Menu > Configure screen > Default buttons > Zoom in (out) > Appearance

**Comment by git2fa** [comment]
> Please update your app to the latest version and adjust the background opacity of Zoom +- buttons in
> 
>     * Menu > Configure screen > Default buttons > Zoom in (out) > Appearance

APP updated now to 5.1.3
Found the option in the german version under Menü > Anzeige konfigurieren > Standard-Schaltflächen > Hineinzommen (Herauszoomen) > Aussehen > Hintergrund-Deckkraft.
Adjusted the opacity to the maximum - but there is no change!
This option only modify the opacity of the + and - for zooming in or out (like seen in the example on the right)!

There is still nearly no contrast for the buttons when set a point and the options will appear.

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/a1231766-fb81-4b97-b3f5-34300edffcec"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by cyclingcat** [comment]
I do not mean any elements on the map but the texts in the general UI (e. g. the settings screens). The "Appearance" settings only changed the transparency of the + and - buttons on the map, at least on my device (Realme 8 Pro, OsmAnd~ v5.1.7 from F-Droid) and doesn't affect the contrast for the texts.

The Android options are global settings; changing them would affect all other apps as well. (On my devices there is a "High contrast" setting, but it stupidly sets the all foreground texts to White in dark mode, dropping _all_ color information, thus making the UI less clear.)

What about the following implementation idea (some kind of a "poor man's theme management"):

* Look for a (XML/JSON/YAML/INI/whatever) file with a given name (e.g. "colors_override.xml") in an app-specific directory a non-root user has write access to.

* If such a file exists, override the (hard-coded?) colours with the ones from this file.

* Document the structure of this file somewhere.

I'm not a "real" software developer (let alone an Android developer), but this sounds quite easy to implement for me. It wouldn't even be necessary to change the settings UI. This would be a huge improvement compared to the effort IMHO, and users could share their colour override files in order to help other users without experience in XML/... editing.

The cycling cat

**Comment by tomFlowee** [comment]
I thought that this was communicated already...

As Cycling Cat states, the contrast is poor.
You can ask web apps to confirm that your artist that picked the colors is not doing their job;
https://coolors.co/contrast-checker/535456-1e1f21

Rated as "very poor".

I don't understand that the OSMAnd people keep insisting it is solved, it is very much impossible to read the buttons and sub texts for normal people outside or under low contrast situations.

This is a direct result of the color palette that OSMAnd ships.

I've given up trying to convince the graphics artist that picked that palette (probably a young guy/gal with very good eyesight),
but the rest of the team should be more honest about the facts. The palette gives really bad readability. As confirmed by independent developers. Search for any other "color contrast" website to repeat the test.

**Comment by dmpr0** [comment]
@tomFlowee Hi! Thanks for you comment. Where did you get those colors? Please try real ones https://coolors.co/contrast-checker/8c8c8c-1e1f20

**Comment by dmpr0** [comment]
@cyclingcat 
Could you please provide screenshots of the places in the app where the text is not contrasting enough?

**Comment by tomFlowee** [comment]
<img width="580" height="309" alt="Image" src="https://github.com/user-attachments/assets/416715ce-3057-4e8d-807a-52aa981dcac7" />

My color picker definitely doesn't show that bright color as your "real ones" show.
My guess is that you got that from the sources, but did you consider that on screen a very thin line of text (it is a small font) effectively bleeds out. You can use your screen picker and unless you're luck, the icons and text have a much less bright value than your "real one".

I quote "real one" because the sources are not the real if it differs from  people actually have in front of them.

To conclude;
outside when the contrast of a normal phone is already less due to bright sunshine.
a small font
a dark gray background
with a light gray foreground.

It is not contrasting enough to be able to read the text.

Please make the 'smaller text' (and icons) have the same white color as the main text.
OR have it be a contrasting color. Blue, yellow, orange. Whatever.

Related issue; https://github.com/osmandapp/OsmAnd/issues/20738
Marked as fixed, but it improved contrast yet did not fix it.

**Comment by git2fa** [comment]
> [@cyclingcat](https://github.com/cyclingcat) Could you please provide screenshots of the places in the app where the text is not contrasting enough?

There are already screenshots here (https://github.com/osmandapp/OsmAnd/issues/9125#issuecomment-3195908115), that show active icons that are so dark like those that are deactivated. Practically I remember the position where I must touch, because it is not possible to see them outside.

Maybe it is possible to criticize that users choose a dark theme for Android, but when you have a phone with a OLED screen you save really much energy using a dark theme and you have a much bigger runtime (of course not relating the coloured maps in OsmAnd).
I can only say that I have no problem with icons in other APP's.

When OsmAnd does not offer different designs, the standard design should be optimized for a good working contrast.
